[refman] Clarify meaning of goal in documentation of instantiate. (#13686)
coq/coq: doc/sphinx/proof-engine/tactics.rst

>    This variant allows to refer to an existential variable which was not named
>    by the user. The :n:`@natural` argument is the position of the existential variable

Reviewer: ```suggestion
   This variant selects an unnamed existential variable by its position.
   The :n:`@natural` argument is the position of the existential variable
```

Does it also allow selecting named existential variables?
If your expression has "?ev1 ... ?ev1 ... ?ev2", is ev2 # 2 or # 3?.
I expect counting begins with 1 as usual.

Author: >Does it also allow selecting named existential variables?

Sure, but it wouldn't be useful to the user.
Counting begins at 1, occurrences are counted *right to left* and multiple occurrences of the same evar are counted multiple times.
I'll try to write all that down.

Reviewer: _**right to left???**_  What is this, Hebrew?  Sounds like a bug to me.  Maybe put that in boldface, it's unexpected.  I didn't notice till you mentioned it.

>    from right to left in the conclusion of the goal (use one of the variants below to select
>    an existential variable in a hypothesis). Because this variant is not robust to slight

Reviewer: ```suggestion
   from right to left in the conclusion of the goal.  (Use one of the variants below to select
   an existential variable in a hypothesis.) Because this variant is not robust to slight
```
